src/material_library.cpp

```diff
@@ -0,0 +1,395 @@
+#include <unistd.h>
+#include <iostream>
+
+#ifndef PYNE_IS_AMALGAMATED
+#include "material_library.h"
+#endif
+
+// Empty Constructor
+pyne::MaterialLibrary::MaterialLibrary(){};
+
+// Default constructor
+pyne::MaterialLibrary::MaterialLibrary(const std::string& file,
+                                       const std::string& datapath) {
+  
+
+  // load materials
+  from_hdf5(file, datapath);
+};
+
+// Append Material to the library from hdf5 file
+void pyne::MaterialLibrary::from_hdf5(const std::string& filename,
+                                      const std::string& datapath) {
+  
+  if (!pyne::file_exists(filename)) {
+    throw std::runtime_error("File " + filename +
+                             " not found or no read permission");
+  }
+  std::string nucpath;
+  std::string full_datapath = datapath;
+
+  // Set file access properties so it closes cleanly
+  hid_t fapl;
+  fapl = H5Pcreate(H5P_FILE_ACCESS);
```

Comment:
```suggestion
  hid_t fapl = H5Pcreate(H5P_FILE_ACCESS);
```
(or this a style violation?)